pkg/download/blob.go

```diff
@@ -121,25 +121,35 @@ func CreateAppendBlob(blobURI, blobSas string) (*storage.Blob, error) {
 		return nil, err
 	}
 
-	fileName := getFileName(blobURI)
+	fileName, blobPathError := getBlobPathAfterContainerName(blobURI, containerRef.Name)
 	if fileName == "" {
-		return nil, fmt.Errorf("cannot extract file name from URL: %q", blobURI)
+		return nil, errors.Wrapf(blobPathError, "cannot extract blob path name from URL: %q", blobURI)
 	}
 
 	blobref := containerRef.GetBlobReference(fileName)
-	err = blobref.PutAppendBlob(nil) // Create the page blob
+	err = blobref.PutAppendBlob(nil) // Create the append blob
 	if err != nil {
 		return nil, err
 	}
 
 	return blobref, nil
 }
 
-// Extract the file name only from the blob uri
-func getFileName(blobURI string) string {
-	s := strings.Split(blobURI, "/")
-	if len(s) > 0 {
-		return s[len(s)-1]
+// Extract the suffix after the container name from blob uri
```

Comment:
consider adding a unit test for this utility method

Reply:
I'll address this part of https://msazure.visualstudio.com/One/_workitems/edit/15140524
Need to take this to GA today.